### DISCUSSION TITLE
Wifi does not connect before login

### DISCUSSION THREAD

**Discussion by RodriguezLeo** [chris1111/Wireless-USB-OC-Big-Sur-Adapter#237] [General]
Hello!

i'm using monterey 12.5 and i can't get the wifi to connect before login.
In previous versions this worked correctly.

Sorry for my English, I hope it is understood correctly

**Comment by chris1111** [comment]
Adapter??

**Reply by RodriguezLeo** [reply]
tp link t2u nano

**Comment by chris1111** [comment]
make Sur setup in OpenCore is correct because this adapter is support

**Reply by RodriguezLeo** [reply]
yes, the adapter works fine.
The problem is that the wifi does not connect automatically until you log in.

for example.
- I turn on the pc. at the login screen, wifi does not connect (try with ssh)
- login with user, wifi works correctly
- I close the user session, at the login screen Wi-Fi continues to work correctly.
- restart pc, wifi is back as in the first step

that worked fine in previous versions.

**Comment by chris1111** [comment]
Try different USB port

**Reply by RodriguezLeo** [reply]
ok, I tried with several ports and with all of them I got the same result
